projects/ngrx.io/content/guide/signals/signal-method.md

```diff
@@ -0,0 +1,200 @@
+    console.log(double);
+  })
+}
+```
+
+`logDoubledNumber` can be called with a static value of type `number`, or a Signal of type `number`:
+
+```ts
+
+@Component({ /* ... */})
+export class NumbersComponent {
+  // 👇 This method will have an input argument
+  // of type `number | Signal<number>`.
+  readonly logDoubledNumber = signalMethod<number>(num => {
+    const double = num * 2;
+    console.log(double);
+  })
+
+  constructor(): void {
+    // 👇 prints 2 synchronously
+    this.logDoubledNumber(1);
+
+    const value = signal(2);
+    // 👇 prints 4 asynchronously (triggered by an internal effect())
+    this.logDoubledNumber(value);
+  }
+}
+```
+
+## Automatic Signal Tracking
+
+`signalMethod` uses an `effect` internally to track the Signal changes.
+
+By default, the `effect` runs in the injection context of the caller. In the example above, that is `NumbersComponent`. That means, that the `effect` is automatically cleaned up when the component is destroyed.
+
+If the call happens outside of an injection context, then the injector of the `signalMethod` is used. This would be the case, if `logDoubledNumber` runs in `ngOnInit`:
+
+```ts
+
+@Component({ /* ... */})
+export class NumbersComponent {
+  readonly logDoubledNumber = signalMethod<number>(num => {
+    const double = num * 2;
+    console.log(double);
+  })
+
+  ngOnInit(): void {
+    this.logDoubledNumber(1);
+
+    const value = signal(2);
+    // 👇 uses the injection context of the `signalMethod`
+    this.logDoubledNumber(value);
+  }
+}
+```
+
+For the `NumbersComponent`, it doesn't make a difference. Again, the `effect` is automatically cleaned up when the component is destroyed.
+
+Careful, when `signalMethod` is used in a service which is provided in `root`:
+
+```ts
+
+@Injectable({providedIn: 'root'})
+export class NumbersService {
+  readonly logDoubledNumber = signalMethod<number>(num => {
+    const double = num * 2;
+    console.log(double);
+  })
+}
+
+@Component({ /* ... */})
+export class NumbersComponent {
+  readonly logDoubledNumber = inject(NumbersService).logDoubledNumber;
+
+  ngOnInit(): void {
+    this.logDoubledNumber(1);
+
+    const value = signal(2);
+    // 👇 uses the injection context of the `NumbersService`, which is root.
+    this.logDoubledNumber(value);
+  }
+}
+```
+
+Here, the `effect` outlives the component, which would produce a memory leak.
+
+As a consequence, try to call the "processor function" always in an injection context:
+
+```ts
+
+@Component({ /* ... */})
+export class NumbersComponent {
+  readonly logDoubledNumber = inject(NumbersService).logDoubledNumber;
+
+  ngOnInit(): void {
+    this.logDoubledNumber(1);
+
+    const value = signal(2);
+    // 👇 uses the injection context of the `NumbersService`, which is root.
+    this.logDoubledNumber(value);
+  }
+}
+```
+
+## Providing an Injector
+
+If you cannot run the "processor function" in an injection context, you can also provide an injector manually:
```

Comment:
for consistency with other pages from `@ngrx/signals` docs, impersonal style is used:

```suggestion
When a `signalMethod` is created in an ancestor injection context, it's necessary to explicitly provide the caller injector to ensure proper cleanup:
```